src/components/RangeSlider.react.js

```diff
@@ -11,6 +11,9 @@ export default class RangeSlider extends Component {
     constructor(props) {
         super(props);
         this.propsToState = this.propsToState.bind(this);
+        this.DashSlider = props.tooltip
+            ? createSliderWithTooltip(Range)
+            : Range;
```

Comment:
This will break if someone starts out with no tooltips and adds them later.

Reply:
OK — I added this back to the render method; that should be fine right?

Reply:
On second thought, putting this in the `render()` method will break dragging behavior (unfortunately my screen recorder can't record a cursor; I'm dragging after hovering on the handle): 

![drag](https://user-images.githubusercontent.com/9068290/59319249-f29f7e80-8c97-11e9-95ee-6fc235b81ecf.gif)

I'm not sure I actually understand the problem with this in the constructor. If I have a hot-reload app, and I add/remove the `tooltip` prop in `Range`/`RangeSlider`, Dash will handle it fine. The only thing that will be reset is any value that has been changed via user interaction (i.e. dragging/clicking the handle).

<details>
  <summary>This is a simple app I'm using to test</summary>

``` python
import dash
import dash_core_components as dcc
import dash_html_components as html

app = dash.Dash(__name__)
server = app.server

app.layout = html.Div([
    html.H1("Tooltips"),
    html.Br(),
    html.H1("Slider, always visible"),
    dcc.Slider(
        id='my-slider',
        min=0,
        max=20,
        step=0.5,
        value=10,
        tooltip={'always_visible': True}
    ),

    html.Br(),

    html.H1("RangeSlider, on hover"),
    dcc.RangeSlider(
        id='my-range-slider',
        min=0,
        max=20,
        step=0.5,
        value=[5,10],
        tooltip={'placement': 'bottom'}
    ),
])

if __name__ == '__main__':
  app.run_server(debug=True)
```
</details>

Reply:
The problem is if `tooltip` is the output of a callback

Reply:
Ah right, of course. I've added the same conditional to `componentWillReceiveProps` and `tooltip` callabacks + normal drag behaviour appear to be working correctly now.